packages/net-stubbing/lib/server/intercepted-request.ts

```diff
@@ -197,8 +201,11 @@ export class InterceptedRequest {
           }
         }
 
-        if (eventName === 'before:request') {
-          if (immediateStaticResponse) {
+        if (eventName === 'before:request' && immediateStaticResponse) {
+          // Since StaticResponse is conflated with InterceptOptions, only send an immediate response if there are keys other than `log`.
```

Comment:
This comment solidified some of what I've felt reading the code here - static response and log are conflated in the code.

It makes sense that they overlap when `.intercept()` is called - it's the cost of how we've overloaded the arguments.

But I wonder if things would be cleaner if "log" were pulled out at that one point - so you'd have a static response / no static response, and then log / don't log to be passed around separately, rather than passing one object being passed around all together and pulled apart into these components as-needed across multiple areas.

Thoughts?

Reply:
Agreed, it's pretty messy this way. I'll update the to break out the options from the `StaticResponse`. How do you feel about the public API combining `log` with `StaticResponse`?

Reply:
Hm. This method signature is heavily overloaded - is it possible to fold it into `routeMatcher` rather than `staticResponse`? That might make more sense at a glance, but I haven't worked through the implications; I'll believe you if you say that's complicated / we can't do it. :rofl:

Reply:
Minor note for when you work on docs - we have some broken links on the `.intercept()` page. https://docs.cypress.io/api/commands/intercept#Syntax- all the links in the Syntax section don't connect properly.

Reply:
One nice thing about having it in the 2nd argument is that it maintains this sort of signature:

```
cy.intercept(which requests you're targeting, what you want to do to those requests)
```

But the 2nd argument is greatly overloaded as you say, so putting it in the first could be nice. However, when we go to add `req.log` and allow dynamically toggling logs in the future, the user could set it in two places:

```
cy.intercept({ log: false }, (req) => req.log = false)
```

...which starts to feel a little weird to me.

Reply:
I worked on ^^ a bit and didn't feel that it was really making the code much clearer. I think this would be worth re-evaluating if we had more `InterceptOptions`, but with just `log` it's a marginal improvement for now.